### PR TITLE
Add example of pro plugin with private npm modules

### PR DESCRIPTION
**Pull request by welteki**
## Description

- Add build example for private npm modules.
- Fix the commands in the python example. They used `faas-cli <command>` instead of `faas-cli pro <command>`.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Add example for building functions using private npm modules.

## How Has This Been Tested?

Tested all instructions and verified changes to the docs visually.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

### CODE REVIEW

File: docs/cli/build.md [hunk 1]
```diff
@@ -133,6 +133,58 @@ RUN --mount=type=secret,id=netrc,mode=0666,dst=/home/app/.netrc \
...
+provider:
+  name: openfaas
+  gateway: http://127.0.0.1:8080
+functions:
+  withprivatenpm:
+    lang: node17
+    handler: ./withprivatenpm
+    image: openfaasltd/withprivatenpm:latest
+    build_secrets:
+      npmrc: /home/welteki/.npmrc
```
Comment (alexellis) [inline docs/cli/build.md:173]: Can you use the string interpolation we talked about on the last community call, and update the previous Python example too?

File: docs/cli/build.md [hunk 2]
```diff
@@ -133,6 +133,58 @@ RUN --mount=type=secret,id=netrc,mode=0666,dst=/home/app/.netrc \
...
+Create a function:
+
+```bash
+export OPENFAAS_PREFIX=openfaasltd
+faas-cli template store pull node17
+faas-cli new --lang python3-http withprivatenpm
+mv withprivatenpm.yml stack.yml
+```
+
+Edit your `~/.npmrc` file to include the authentication token for your npm registry.
```
Comment (alexellis) [inline docs/cli/build.md:154]: Isn't there an npm login command for this?

What about a section with the instructions you used for GHCR?

File: docs/cli/build.md [hunk 3]
```diff
@@ -133,6 +133,58 @@ RUN --mount=type=secret,id=netrc,mode=0666,dst=/home/app/.netrc \
...
+      npmrc: ${HOME}/.npmrc
+```
+
+Run a build with:
+
+```bash
+faas-cli pro build -f stack.yml
+```
+
+You'll also need an updated version of the node template where we tell buildkit to mount the secret passed in from the OpenFaaS Pro plugin. Update `template/node17/Dockerfile` and replace the second `npm i` command with:
```
Comment (alexellis) [inline docs/cli/build.md:182]: Let's not mention buildkit here, Id on't think it's relevant. If I did in the previous section, remove it there too.

Comment (welteki) [reply docs/cli/build.md]: I removed the buildkit mention here and in the python example.

File: docs/cli/build.md (hunk 2 above)
Comment (welteki) [reply docs/cli/build.md]: I linked to the instructions for both the GithHub Packages registry and the npmjs registry.

**Review by alexellis — APPROVED** [review]
Approved

**Pull request merged**